pandas/core/generic.py

```diff
@@ -4061,70 +4069,86 @@ def sort_values(
 
              .. versionadded:: 1.0.0
 
+        key : callable, optional
+            Apply the key function to the values
+            before sorting. This is similar to the `key` argument in the
+            builtin :meth:`sorted` function, with the notable difference that
+            this `key` function should be *vectorized*. It should expect a
+            ``Series`` and return a Series with the same shape as the input.
+            It will be applied to each column in `by` independently.
```

Comment:
It's not clear to me whether it gets the full dataframe, or only a subset of the dataframe for the columns the user specified to sort by.

Reply:
Only the subset of the columns selected in `by`.